specifications/Moving_Cat_Puzzle/Cat.tla

```diff
@@ -0,0 +1,99 @@
+You have N boxes numbered 1 to N, in one of which a cat is
+hiding. Every night, she moves to an adjacent box, and every morning,
+you have one chance to inspect a box to find her. How do you find the
+cat?
+
+Solution: You start at box 2 and move one by one to N - 1. Then you
+observe N - 1 again and start moving back to box 2. Repeat and
+eventually you will find the cat. E.g with 5 boxes you check 2, 3, 4,
+4, 3, 2; with 4 boxes you check 2, 3, 3, 2.
+
+There is a simpler solution if the number of boxes is odd. The
+solution here works in both even and odd cases.
+
+The puzzle is trivial / impossible for a single box, since the cat
+cannot move.
+
+---- MODULE Cat ----
+
+EXTENDS Naturals
+
+CONSTANTS
+  Number_Of_Boxes
+
+ASSUME Number_Of_Boxes >= 2
+
+VARIABLES
+  \* The box the cat is currently hiding in
+  cat_box,
+
+  \* The box that is observed
+  observed_box,
+
+  \* The direction in which our search is progressing
+  direction
+
+vars == <<cat_box, observed_box, direction>>
+
+----------------------------------------------------------------------
+
+\* Initially the cat is in an arbitrary box. We start our search at
+\* box 2, moving to the right.  It doesn't actually matter where you
+\* start as long as you eventually fall into the pattern described by
+\* the solution.
+Init ==
+  /\ cat_box \in 1 .. Number_Of_Boxes
+  /\ observed_box = 2
```

Comment:
Does it always have to be box 2 regardless of `Number_Of_Boxes`?

Reply:
No - fixed that to start anywhere in the search pattern.